invenio_jobs/services/services.py

```diff
@@ -114,8 +116,12 @@ def search(self, identity, params):
                 ]
             )
 
+        # if filters == [] -> True
+        # if filters != [] -> False
+        default_element = not bool(len(filters))
```

Comment:
Ouch, I've seen this footgun before 😬 : https://github.com/inveniosoftware/invenio-communities/commit/9a6e4b1f23112739c92bbea106b466a8258737b6

Not super happy with the solution, I would prefer something like what I did in the commit above, i.e. have an `if filters: ...` which modifies some base query accordingly... In general, though I think we need to check all the places we `sa.or_(...)` things dynamically (or find a smarter way to avoid this in general).

Reply:
I've pushed a commit with a bit simpler logic IMO on handling the filtering. 

It doesn't find a "smarter" way to handle this type of situation in general (I think "avoiding" it is the best way ;) ), but unblocks this PR.